libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -164,3 +184,19 @@ vector<ASTString> DeclarationContainer::similarNames(ASTString const& _name) con
 
 	return similar;
 }
+
+void DeclarationContainer::populateHomonyms(back_insert_iterator<Homonyms> _it) const
+{
+	for (DeclarationContainer const* innerContainer: m_innerContainers)
+		innerContainer->populateHomonyms(_it);
+
+	for (auto [name, location]: m_homonymCandidates)
+	{
+		Declaration const* homonymDeclaration = nullptr;
+		for (auto const* declaration: m_enclosingContainer->resolveName(name, true, true))
+			homonymDeclaration = declaration;
```

Comment:
Any reason for not adding a `break` after the first?

Reply:
Historical reasons, it is what we did before. But I wondered too :).

Actually, I thought it might make more sense to process all declarations, not just first or last. I did not want to address it within this PR, but since you brought it, please look at 27ffffb1952d087295b31e5e44e12a33f4ca1dd1.

Reply:
I think it's better to only report one shadowed declaration.

Reply:
Why? It makes code slightly shorter and avoids the decision which name collision to report.